test/util/deterministic.cpp

```diff
@@ -0,0 +1,108 @@
+#include <fstream>
```

Comment:
Why the name deterministic.cpp/h? Seems like graph_verifier.cpp/h might fit the content better?